### ISSUE TITLE
type-comparison (E721) doesn't work when comparing against an actual type

### ISSUE THREAD

**Issue by DetachHead** [astral-sh/ruff#6260]
```py
foo = 1
if type(foo) is int: # no error
    ...
if type(foo) is type(1): # error
    ...
```
the pylint rule `unidiomatic-typecheck` catches both cases

**Comment by KotlinIsland** [comment]
@charliermarsh Could you please clarify if the change here applies to only builtin types? And if so why, because in pylint it applies to any type, not just builtins:

```py
class A: ...
a = A()
type(a) == A  # C0123: Use isinstance() rather than type() for a typecheck. (unidiomatic-typecheck)
```

**Comment by charliermarsh** [comment]
I believe that allowing arbitrary expressions on the right-hand side will lead to false positives for legitimate usages, as in: https://github.com/PyCQA/pycodestyle/issues/882#issuecomment-588480243.

**Comment by KotlinIsland** [comment]
Oh, is this because Ruff isn't able to tell if the expression on the rhs is a `type` or not?

**Comment by charliermarsh** [comment]
Yes that's correct. E.g., this is contrived but seems reasonable?

```python
type_of_b = type(b)
type(a) is type_of_b
```

Pylint flags this under `unidiomatic-typecheck`, but it seems fine to me.

**Comment by KotlinIsland** [comment]
> Yes that's correct.

If Ruff can't determine if the rhs is a `type` then I understand that it's better to be less strict here.

 E.g., this is contrived but seems reasonable?
> 
> ```python
> type_of_b = type(b)
> type(a) is type_of_b
> ```
> 
> Pylint flags this under `unidiomatic-typecheck`, but it seems fine to me.

I would write:
```py
type_of_b = type(b)
isinstance(a, type_of_b)
```
Assuming the context is that I'm just dealing with some kind of usual operation where what I really want is a compatible subtype, and that I don't want to be checking for the exact same type.

Isn't the whole point of the inspection to encourage `isinstance` over `==`/`is` for types because `isinstance` is aware of subtypes but `==`/`is` isn't?
```py
isinstace(True, int)  # True
type(True) == type(int)  # False
```

**Comment by hauntsaninja** [comment]
Would you accept a PR with the naive autofix for this?

**Comment by charliermarsh** [comment]
Yeah, we can make it a suggested fix.

(Also open to further refinements on the rule logic itself, I’m not certain that our criteria is optimal given above discussion.)
